SU2_CFD/include/numerics/species/species_sources.hpp

```diff
@@ -89,4 +89,52 @@ class CSourceAxisymmetric_Species : public CSourceBase_Species {
    * \return Lightweight const-view of residual and Jacobian.
    */
   ResidualType<> ComputeResidual(const CConfig* config) override;
+
+};
+
+/*!
+ * \class CSourcePieceWise_transportedScalar_general
+ * \brief Class for integrating the source terms of the transported scalar turbulence model equations.
+ * \ingroup SourceDiscr
+ * \author N. Beishuizen
+ */
+class CSourcePieceWise_transportedScalar_general final : public CNumerics {
+
+private:
+  su2double *Residual = nullptr;
+  su2double **Jacobian_i = nullptr;
+  su2double *scalar_sources = nullptr;
+   bool incompressible;
+
+public:
+  /*!
+   * \brief Constructor of the class.
+   * \param[in] val_nDim - Number of dimensions of the problem.
+   * \param[in] val_nVar - Number of variables of the problem.
+   * \param[in] config - Definition of the particular problem.
+   */
+  CSourcePieceWise_transportedScalar_general(unsigned short val_nDim, unsigned short val_nVar, const CConfig* config);
+
+  /*!
+   * \brief Residual for source term integration.
+   * \param[in] config - Definition of the particular problem.
+   * \return A lightweight const-view (read-only) of the residual/flux and Jacobians.
+   */
+  ResidualType<> ComputeResidual(const CConfig* config) override;
+
+  /*!
+   * \brief Set the value of the progress variable source term for the flamelet model.
+   * \param[in] val_sourcepv_i - Value of the source term at point i.
+   * \param[in] val_sourcepv_j - Value of the source term at point j.
+   */
+  inline void SetScalarSources(su2double *val_scalar_sources) override {
+    for (auto i_var=0u; i_var < nVar; i_var++)
+      scalar_sources[i_var] = val_scalar_sources[i_var];
+  }
```

Comment:
Is the comment correct?
Can you just pass a const su2double* instead of copying the data? (like we do with primitives etc)

Reply:
Thanks for the scrutiny :-) It looks like this comment was copied from elsewhere. And yes, I think we can make this a const su2double*, I will go through the functions for more consting.